### PR TITLE
Add support for @npm: aliases

### PR DESCRIPTION
**Pull request by chmeliik**
https://docs.npmjs.com/cli/v8/commands/npm-install

See `npm install <alias>@npm:<name>`

If an aliased package appears in the lockfile, the Package.name
attribute will now properly report the name rather than the alias.
The new Package.alias attribute reports the alias.

Signed-off-by: Adam Cmiel <acmiel@redhat.com>

### CODE REVIEW

**Comment by chmeliik** [comment]
PTAL, with this fix Cachito can finally process https://github.com/xperimental/kibana :its-so-beautiful:

(tested with https://github.com/containerbuildsystem/cachito/pull/836)

File: pyarn/lockfile.py [hunk 1]
```diff
@@ -59,9 +73,21 @@ def from_dict(cls, raw_name, data):
             checksum=data.get("integrity"),
             relpath=path,
             dependencies=data.get("dependencies", {}),
+            alias=alias,
         )
 
 
+def _removeprefix(s: str, prefix: str) -> str:
```
Comment (brunoapimentel) [inline pyarn/lockfile.py]: Nitpick: I think this is missing an underscore, `_remove_prefix`

**Review by brunoapimentel — APPROVED** [review]
Thanks for digging yet another Yarn behavior and making Pyarn compliant :pray:

**Review by lkolacek — APPROVED** [review]
Great job Adam :+1:

**Pull request merged**